mtail_violations_exporter/gatekeeper.mtail

```diff
@@ -0,0 +1,45 @@
+# model the gauge based on labels
+gauge opa_violations by constraint_kind, constraint_name, resource_kind, resource_namespace, resource_name, msgid, context
+
+# temporary variable to store message id
+text msgid 
+
+# temporary variable to store message context
+text context
+
+# pattern for opa violation attributes
+const VIOLATION /{.*\\"constraint_kind\\"\s*:\s*\\"(?P<constraint_kind>.*?)\\".*\\"constraint_name\\"\s*:\s*\\"(?P<constraint_name>.*?)\\".*\\"resource_kind\\"\s*:\s*\\"(?P<resource_kind>.*?)\\".*\\"resource_namespace\\"\s*:\s*\\"(?P<resource_namespace>.*?)\\".*\\"resource_name\\"\s*:\s*\\"(?P<resource_name>.*?)\\".*}/
```

Comment:
It looks like Golang's built in JSON marshaler does have stably sorted map keys:

https://golang.org/pkg/encoding/json/#Marshal

So this looks like it should work. Care should be taken here though, as adding new fields will break this scraper.

If it's possible to semantically process the JSON (e.g. actually convert this string to a JSON object and pull out values of interest), then this would be less brittle against the adding of new fields to the log line (which would be considered a backwards compatible change).

Reply:
Mtail is an unstructured parser, so no semantic JSON processing.

If you add a new field, these should still be sorted and it will just ignore your new field as it does today with `constraint_action`. 

That said I think it *would* be better to break out each of these fields into their own const & not rely on the sorting. Does this work for now for you & we can break apart this VIOLATION parse the next time we update this program?